src/mempool.h

```diff
@@ -46,12 +54,31 @@ class MemoryPool {
     struct ibv_pd* pd_;
 };
 
+typedef struct {
+    MemoryPool* pool_ptr = NULL;
+    struct ibv_pd* pd = NULL;
+    std::vector<MemoryPool*>* mempools = NULL;
```

Comment:
if we move the ALLOCATE New mempool logics to infinistore.cpp, we could use global MM's new method "add Mempool" so we do not need mempools anymore.